### PR TITLE
Verify if script name provided is a file before running it

### PR DESCRIPTION
**Pull request by dukecat0**
- [x] I have added an entry to `docs/changelog.md`

## Summary of changes
`pipx run`: Verify whether the script name provided is a file before running it
Closes https://github.com/pypa/pipx/issues/1152

## Test plan

Tested by running

```
nox -s tests-3.12
```

### CODE REVIEW

**Comment by chrysle** [comment]
Didn't you want to detect whether the file is a valid script?

File: src/pipx/commands/run.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def maybe_script_content(app: str, is_path: bool) -> Optional[str]:
 
     # Look for a local file first.
     app_path = Path(app)
-    if app_path.exists():
+    if app_path.exists() and app_path.is_file():
```
Comment (uranusjr) [inline src/pipx/commands/run.py:51]:
```suggestion
    if app_path.is_file():
```

I believe `is_file` implies `exists`.

https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_file

> False is also returned if the path doesn’t exist […].

**Comment by dukecat0** [comment]
> Didn't you want to detect whether the file is a valid script?

I think a file that has no `.py` extension or shebang can be a valid script as well. (At least the Python interpreter doesn't care about it.) That's why I personally believe that checking whether it is a file is enough.

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Test please 👍

**Comment by dukecat0** [comment]
Not sure if we need a test for this...? I personally think this is a very minor bug and doesn't require a test.

**Comment by gaborbernat** [comment]
> Not sure if we need a test for this...? I personally think this is a very minor bug and doesn't require a test.

I disagree.

**Review by gaborbernat — CHANGES_REQUESTED** [review]
Tests please.

**Comment by dukecat0** [comment]
> Tests please.

Added

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**